examples/jenkins/pipeline/pipelinetemplate.json

```diff
@@ -0,0 +1,529 @@
+{
+  "kind": "Template",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "ruby-helloworld-sample",
+    "creationTimestamp": null,
+    "annotations": {
+      "description": "This example shows how to create a simple ruby application in openshift origin v3",
+      "iconClass": "icon-ruby",
+      "tags": "instant-app,ruby,mysql"
+    }
+  },
+  "objects": [
+    {
+      "kind": "BuildConfig",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "sample-pipeline",
+        "creationTimestamp": null,
+        "labels": {
+          "name": "sample-pipeline"
+        }
+      },
+      "spec": {
+        "triggers": [
+          {
+            "type": "GitHub",
+            "github": {
+              "secret": "secret101"
+            }
+          },
+          {
+            "type": "Generic",
+            "generic": {
+              "secret": "secret101"
+            }
+          }
+        ],
+        "source": {
+          "type": "Git",
+          "git": {
+            "uri": "https://github.com/openshift/ruby-hello-world"
+          }
+        },
+        "strategy": {
+          "type": "JenkinsPipeline",
+          "jenkinsPipelineStrategy": {
+            "jenkinsfile": "node('agent') {\nstage 'build'\ndef builder = new com.openshift.jenkins.plugins.pipeline.OpenShiftBuilder(\"\", \"ruby-sample-build\", \"pipelineproject\", \"\", \"\", \"\", \"\", \"true\", \"\", \"\")\nstep builder\nstage 'deploy'\ndef deployer = new com.openshift.jenkins.plugins.pipeline.OpenShiftDeployer(\"\",\"frontend\",\"pipelineproject\",\"\",\"\")\nstep deployer\n}"
+          }
+        }
+      }
+    },
+    {
+      "kind": "Service",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "frontend",
+        "creationTimestamp": null
```

Comment:
Shouldn't be wrapped in "metadata" - it is just []api.ObjectReference.

On May 23, 2016, at 7:01 PM, Sam Padgett <notifications@github.com> wrote:

In examples/jenkins/pipeline/pipelinetemplate.json
<https://github.com/openshift/origin/pull/8973#discussion_r64304059>:

> +          }
> +        },
> +        "strategy": {
> +          "type": "JenkinsPipeline",
> +          "jenkinsPipelineStrategy": {
> +            "jenkinsfile": "node('agent') {\nstage 'build'\ndef builder = new com.openshift.jenkins.plugins.pipeline.OpenShiftBuilder(\"\", \"ruby-sample-build\", \"pipelineproject\", \"\", \"\", \"\", \"\", \"true\", \"\", \"\")\nstep builder\nstage 'deploy'\ndef deployer = new com.openshift.jenkins.plugins.pipeline.OpenShiftDeployer(\"\",\"frontend\",\"pipelineproject\",\"\",\"\")\nstep deployer\n}"
> +          }
> +        }
> +      }
> +    },
> +    {
> +      "kind": "Service",
> +      "apiVersion": "v1",
> +      "metadata": {
> +        "name": "frontend",
> +        "creationTimestamp": null

Can we add the annotation suggested by @smarterclayton
<https://github.com/smarterclayton> ? The web console is expecting this:

        "annotations": {
          "service.alpha.openshift.io/dependencies": "[{\"metadata\":
{\"name\": \"database\", \"namespace\": \"\", \"kind\":
\"Service\"}}]"
        }

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/8973/files/8a5e60bb95f219933f0917356280bb140ed4ba4f#r64304059>